api/v1alpha1/nodepool_types.go

```diff
@@ -274,7 +274,26 @@ type RollingUpdate struct {
 
 // InPlaceUpgrade specifies an upgrade strategy which upgrades nodes in-place
 // without any new nodes being created or any old nodes being deleted.
-type InPlaceUpgrade struct{}
+type InPlaceUpgrade struct {
+	// MaxUnavailable is the maximum number of nodes that can be unavailable
+	// during the update.
+	//
+	// Value can be an absolute number (ex: 5) or a percentage of desired nodes
+	// (ex: 10%).
+	//
+	// Absolute number is calculated from percentage by rounding down.
+	//
+	// Defaults to 1.
```

Comment:
so should we `// +kubebuilder:default:=1` here?

Reply:
I can definitely do this. The reason I set it up as it stands today was that it matches the MCO behaviour, where maxUnavailable is optional and has no default value, and we "assume" it to be 1 unless the user provides otherwise.

Although I think maybe that makes less sense in the Hypershift context, what do you think?